### PR TITLE
[DUOS-2559][risk=no] Extract catalog styles into a module

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
Partial support for https://broadworkbench.atlassian.net/browse/DUOS-2559

### Summary
Pull a bunch of problematic global styles into a css module for the Dataset Catalog. This PR doesn't try to pull ALL styles into the module, just as many of the problematic ones as possible.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by hams7504 — APPROVED** [review]
👍

File: src/pages/DatasetCatalog.module.css [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+.checkbox[disabled] .regular-checkbox {
+  cursor: not-allowed !important;
+  opacity: 0.65;
+}
+
+.checkbox[disabled] .regular-checkbox:before  {
+  background-color: #eee;
+  border: 1px solid #ccc;
+}
```
Comment (fboulnois) [inline src/pages/DatasetCatalog.module.css:23]: Where do these original styles that are not in the css index come from?

**Review by fboulnois — COMMENTED** [review]
One question:

File: src/pages/DatasetCatalog.module.css (hunk 1 above)
Comment (rushtong) [reply src/pages/DatasetCatalog.module.css:23]: All styles in this module were copied from `index.css`. Not all of them were removed from `index.css`, but the ones defined here will only ever be used by this page. We could use a comprehensive cleanup of `index.css` but the goal of this PR was to unblock your work.

**Review by hams7504 — APPROVED** [review]

**Review by fboulnois — APPROVED** [review]
👍

**Pull request merged**